StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/UsingCodeFixProvider.cs

```diff
@@ -0,0 +1,642 @@
+                if (usingList.Count > 0)
+                {
+                    // sort the original usings on Span.Start, in order to have the correct replace mapping.
+                    usingList.Sort(CompareSpanStart);
+
+                    var indentationSteps = IndentationHelper.GetIndentationSteps(indentationOptions, usingList[0].Parent);
+                    if (usingList[0].Parent is NamespaceDeclarationSyntax)
+                    {
+                        indentationSteps++;
+                    }
+
+                    var indentation = IndentationHelper.GenerateIndentationString(indentationOptions, indentationSteps);
+
+                    var modifiedUsings = usingsHelper.GenerateGroupedUsings(usingList, indentation, false);
+
+                    for (var i = 0; i < usingList.Count; i++)
+                    {
+                        replaceMap.Add(usingList[i], modifiedUsings[i]);
+                    }
+                }
+            }
+        }
+
+        private static void BuildReplaceMapForConditionalDirectives(UsingsHelper usingsHelper, Dictionary<UsingDirectiveSyntax, UsingDirectiveSyntax> replaceMap, IndentationOptions indentationOptions, DirectiveSpan rootSpan)
+        {
+            foreach (var childSpan in rootSpan.Children)
+            {
+                var originalUsings = usingsHelper.GetContainedUsings(childSpan);
+                if (originalUsings.Count > 0)
+                {
+                    // sort the original usings on Span.Start, in order to have the correct replace mapping.
+                    originalUsings.Sort(CompareSpanStart);
+
+                    var indentationSteps = IndentationHelper.GetIndentationSteps(indentationOptions, originalUsings[0].Parent);
+                    if (originalUsings[0].Parent is NamespaceDeclarationSyntax)
+                    {
+                        indentationSteps++;
+                    }
+
+                    var indentation = IndentationHelper.GenerateIndentationString(indentationOptions, indentationSteps);
+
+                    var modifiedUsings = usingsHelper.GenerateGroupedUsings(childSpan, indentation, false);
+
+                    for (var i = 0; i < originalUsings.Count; i++)
+                    {
+                        replaceMap.Add(originalUsings[i], modifiedUsings[i]);
+                    }
+                }
+
+                BuildReplaceMapForConditionalDirectives(usingsHelper, replaceMap, indentationOptions, childSpan);
+            }
+        }
+
+        private static int CompareSpanStart(UsingDirectiveSyntax left, UsingDirectiveSyntax right)
+        {
+            return left.SpanStart - right.SpanStart;
+        }
+
+        private static SyntaxNode AddUsingsToNamespace(SyntaxNode newSyntaxRoot, UsingsHelper usingsHelper, string usingsIndentation, bool hasConditionalDirectives)
+        {
+            var rootNamespace = ((CompilationUnitSyntax)newSyntaxRoot).Members.OfType<NamespaceDeclarationSyntax>().First();
```

Comment:
:bulb: Linq will cause boxing. Seems like a good canidate for a new method in `SyntaxTriviaListEnumerable`.